bears/vcs/git/GitCommitBear.py

```diff
@@ -203,3 +240,74 @@ def check_body(self, body,
                                'Commit body lines should not exceed {} '
                                'characters.'.format(body_line_length))
+
+    def check_issue_reference(self, body,
+                              body_close_issue: bool=False,
+                              body_close_issue_full_url: bool=False,
+                              body_close_issue_on_last_line: bool=False):
+        """
+        Check for matching issue related references and URLs.
+
+        :param body:
+            Body of the commit message of HEAD.
+        :param body_close_issue:
+            Whether to check for the presence of issue reference within
+            the commit body by retrieving host information from git
+            configuration. GitHub and GitLab support auto closing issues with
+            commit messages. Checks for matching keywords in the commit body.
+            By default, if none of ``body_close_issue_full_url`` and
+            ``body_close_issue_on_last_line`` are enabled, checks for presence
+            of short references like ``closes #213``. Otherwise behaves
+            according to other chosen flags.
+            More on keywords follows.
+            [GitHub](https://help.github.com/articles/closing-issues-via-commit-messages/)
+            [GitLab](https://docs.gitlab.com/ce/user/project/issues/automatic_issue_closing.html)
+        :param body_close_issue_full_url:
+            Checks the presence of issue close reference with a full URL
+            related to some issue. Works along with ``body_close_issue``.
+        :param body_close_issue_on_last_line:
+            When enabled, checks for issue close reference presence on the
+            last line of the commit body. Works along with
+            ``body_close_issue``.
+        """
+        if not body_close_issue:
+            return
+
+        host = self.get_host_from_remotes()
+        if host not in self.SUPPORTED_KEYWORD_REGEX:
+            return
+
+        if body_close_issue_on_last_line:
+            body = body.splitlines()[-1]
+            result_message = ('Body of HEAD commit does not contain any {} '
+                              'reference in the last line.')
+        else:
+            result_message = ('Body of HEAD commit does not contain any {} '
+                              'reference.')
+
+        if body_close_issue_full_url:
+            result_message = result_message.format('full issue')
+            issue_keyword_regex = r'https?://{}\S+/issues/\w+'.format(
+                re.escape(host))
+        else:
+            result_message = result_message.format('issue')
+            issue_keyword_regex = r'#\w+'
+
+        concat_keywords = [',', 'and']
+        concat_regex = '|'.join(re.escape(kw) for kw in concat_keywords)
+        joint_regex = r'({1}(?:\s*(?:{0})\s*{1})*)'.format(concat_regex,
+                                                           issue_keyword_regex)
```

Comment:
But using `\w+` would not match `https://github.com/u/r/issues/312`, right?

In fact, I tried many combinations like `\S+`, but couldn't figure out a way to match the entire URL.